staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/utils.go

```diff
@@ -199,3 +200,36 @@ func VerifyVolumePathsForVMDevices(vmDevices object.VirtualDeviceList, volPaths
 	}
 
 }
+
+// isvCenterDeprecated takes vCenter version and vCenter API version as input and return true if vCenter is deprecated
+func isvCenterDeprecated(vCenterVersion string, vCenerAPIVersion string) (bool, error) {
+	minvcversion, err := semver.New(MinvCenterVersion)
+	vcdeprecated := false
+	if err != nil {
+		return false, fmt.Errorf("failed to get parse vCenter version: %s. err: %+v", MinvCenterVersion, err)
+	} else {
+		vcversion, err := semver.New(vCenterVersion)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse vCenter version: %s. err: %+v", vCenterVersion, err)
+		} else {
+			result := vcversion.Compare(*minvcversion)
+			if result == -1 {
+				// vcversion is less than minvcversion
+				klog.Warningf("vSphere version: %s is deprecated. Please consider upgrading vSphere to 6.7u3 or higher", vCenerAPIVersion)
```

Comment:
Either caller of `isvCenterDeprecated` shouldn't log the warning or if warning message needs to be more nuanced this is the only function that should log the warnings.

Reply:
Fixed. Only the caller will print this warning message once.